scripts/platform.cake

```diff
@@ -0,0 +1,73 @@
+#load "runhelpers.cake"
+
+using System;
+
+public class Platform
```

Comment:
it's nice that we have robust platform information captured in one place. Maybe for consistency we can replace the calls to the built-in Cake `IsRunningOnWindows()` global methods used in several places with `platform.IsWindows`. This way we have a guarantee the two logics would not diverge